lib/features/map_view/controllers/map_data_controller.dart

```diff
@@ -1,23 +1,22 @@
+import "package:fast_immutable_collections/fast_immutable_collections.dart";
 import "package:riverpod_annotation/riverpod_annotation.dart";
 
 import "controllers_set.dart";
 
 mixin MapDataController<T extends GoogleNavigable>
-    on AutoDisposeAsyncNotifier<Iterable<T?>?> {
+    on AutoDisposeAsyncNotifier<IList<T?>?> {
   String _textFieldFilterText = "";
   late final MapControllers<T> mapControllers;
 
   @override
-  FutureOr<Iterable<T?>?> build() async {
+  FutureOr<IList<T?>?> build() async {
     final itemSelected = ref.watch(mapControllers.activeMarker);
     if (itemSelected != null) {
-      return [itemSelected]; // shows only selected building
+      return [itemSelected].lock; // shows only selected building
```

Comment:
Tell me if i properly understand that; if the item is selected, the list cannot be modified?

Reply:
I believe `.lock` is just a shortcut for `.toIList()` but I didn't dive deeply